Expensify/App
# #4517 update timezone on user activity

### Details


This PR replaces the original automatic timezone implementation that only checked timezone on app start with a new one, that also checks for updates on specific user activity: send chat message and open profile settings (the 2 places where you will see your current timezone impact rendering).

### Fixed Issues

$ https://github.com/Expensify/App/issues/4517

### Tests


These are long, but are required to check that the timezone is updated:
* on app start
* on message send
* on profile settings open

1. Set timezone to automatic
1. Change [throttle timer](https://github.com/Expensify/App/pull/4799/files#diff-51eb3f80fcb6b3d0666f4a25c1611e373f886291ad7f60f24bad5fc2405c072aR126) to something more testable (5 seconds for example)
2. Open a chat with a second account that is in the same timezone as your active account
3. You should not see participant's local time because you're in the same timezone
4. Change the current timezone in your OS
5. Send a message
6. Participant's local time should appear
7. Change the timezone back
8. Send another message after a throttle timeout passed
9. Participant's local time should disappear
10. Change the timezone again
11. Refresh the page or restart the app
12. Open chat and you see participant's local time again
13. Change the timezone back
14. Open your profile's settings
15. You should see the newly selected timezone there

### QA Steps

1. Set timezone to automatic
2. Open a chat with a second account that is in the same timezone as your active account
3. You should not see participant's local time because you're in the same timezone
4. Change the current timezone in your OS
5. Send a message
6. Participant's local time should appear
7. Change the timezone back
8. Send another message 10 minutes later
9. Participant's local time should disappear
10. Change the timezone again
11. Refresh the page or restart the app
12. Open chat and you see participant's local time again
13. Change the timezone back
14. Open your profile's settings
15. You should see the newly selected timezone there
### Tested On


### Screenshots


#### Web


https://user-images.githubusercontent.com/64093836/130581595-31015651-5ca7-4639-846d-a32e4f6d5ea1.mp4


#### Mobile Web


https://user-images.githubusercontent.com/64093836/130618247-3b1768c0-e467-415a-9d68-e089504b363e.mp4


#### Desktop


https://user-images.githubusercontent.com/64093836/130603509-fcc18965-2c4d-4a6f-bbd5-fa1e0be07443.mp4


#### iOS


https://user-images.githubusercontent.com/64093836/130618137-fd1dbbab-826d-418a-a0f0-fdb453fbdaad.mp4


#### Android

Sorry for the crash in the middle of the video, reloading with hermes results in crash like 50% of the time for me. Still trying to figure it out

https://user-images.githubusercontent.com/64093836/130581926-c4fe3cd7-d6ae-4d86-85f8-42867d139bba.mp4
